src/main/scala/com/danielasfregola/twitter4s/TwitterRestClient.scala

```diff
@@ -62,23 +63,50 @@ trait RestClients
     with TwitterUserClient
 
 object TwitterRestClient {
-
   def apply(): TwitterRestClient = {
-    val consumerToken = ConsumerToken(key = consumerTokenKey, secret = consumerTokenSecret)
-    val accessToken = AccessToken(key = accessTokenKey, secret = accessTokenSecret)
-    apply(consumerToken, accessToken)
+    // See what we have to work with
+    authMode match {
+      case UseOAuth1 => {
+        val consumerToken = ConsumerToken(key = consumerTokenKey.get, secret = consumerTokenSecret.get)
+        val accessToken = AccessToken(key = accessTokenKey.get, secret = accessTokenSecret.get)
+        apply(consumerToken, accessToken)
+      }
+      case UseOAuthBearerToken => {
+        apply(bearerToken = bearerToken.get)
+      }
+      case MixedAuth => {
+        val consumerToken = ConsumerToken(key = consumerTokenKey.get, secret = consumerTokenSecret.get)
+        val accessToken = AccessToken(key = accessTokenKey.get, secret = accessTokenSecret.get)
+        apply(consumerToken, accessToken, bearerToken = bearerToken.get)
+      }
+      case _ => {
+        // Uh, we somehow have no authmode?
+        throw new RuntimeException("TwitterRestClient: No authentication mode is defined")
+      }
+    }
   }
 
+  // Mixed constructor
+  def apply(consumerToken: ConsumerToken, accessToken: AccessToken, bearerToken: String): TwitterRestClient =
+    new TwitterRestClient(consumerToken, accessToken, bearerToken, MixedAuth)
+
+  // OAuth1 constructor
   def apply(consumerToken: ConsumerToken, accessToken: AccessToken): TwitterRestClient =
-    new TwitterRestClient(consumerToken, accessToken)
+    new TwitterRestClient(consumerToken, accessToken, bearerToken = null, UseOAuth1)
+
+  // Bearer token constructor
+  def apply(bearerToken: String) = {
+    new TwitterRestClient(null, null, bearerToken, UseOAuthBearerToken)
```

Comment:
Another sign that the design of the solution is not ideal: we should avoid `null`